lib/matplotlib/colors.py

```diff
@@ -529,7 +529,18 @@ def __call__(self, X, alpha=None, bytes=False):
         if vtype == 'scalar':
             rgba = tuple(rgba[0, :])
         return rgba
-
+    
+    def __copy__(self):
+        """Create new object with the same class and update attributes
+	If object is initialized, copy the elements of _lut list
+	"""
+	cls = self.__class__
```

Comment:
I have bench marked the 2 functions.
The following code is used:
```python
import matplotlib.pyplot as plt
import time

cm = plt.cm.Reds

start = time.time()
cm2 = cm.copy_1()     #Using low level python
end1 = time.time()
cm3 = cm.copy_2()     #Copying the attributes manually
end2 = time.time()
print("Method 1 :" + str(end1-start))
print("Method 2 :" + str(end2-end1))
```

I get the following output:
Method 1 :1.8358230590820312e-05
Method 2 :2.6941299438476562e-05

I can definitely say that using low level python functions would lead to a simpler, faster code, which might be a little difficult to understand by seeing it.

#### Test
I've used the copy_1 function to copy the object.
```python
import numpy as np
import matplotlib.pyplot as plt

cm = plt.cm.Reds
cm([-1, 0, .5, np.nan, np.inf])

cm.set_bad('y')
cm2 = cm.copy_1()
cm2.set_bad('b')
print(cm._rgba_bad)
print(cm2._rgba_bad)

cm.set_under('y')
cm2.set_under('r')
print(cm._rgba_under)
print(cm2._rgba_under)

cm.set_over('c')
cm2.set_over('m')
print(cm._rgba_over)
print(cm2._rgba_over)
```

Output is as follows:
(0.75, 0.75, 0.0, 1)
(0.0, 0.0, 1.0, 1)
(0.75, 0.75, 0.0, 1)
(1.0, 0.0, 0.0, 1)
(0.0, 0.75, 0.75, 1)
(0.75, 0.0, 0.75, 1)

As we can see, using the function on both Colormaps doesn't change the other